plugins/repository-s3/src/main/java/org/opensearch/repositories/s3/S3BlobContainer.java

```diff
@@ -229,34 +246,29 @@ private void doDeleteBlobs(List<String> blobNames, boolean relative) throws IOEx
-                AmazonClientException aex = null;
+                SdkException aex = null;
                 for (DeleteObjectsRequest deleteRequest : deleteRequests) {
-                    List<String> keysInRequest = deleteRequest.getKeys()
+                    List<String> keysInRequest = deleteRequest.delete()
+                        .objects()
                         .stream()
-                        .map(DeleteObjectsRequest.KeyVersion::getKey)
+                        .map(ObjectIdentifier::key)
                         .collect(Collectors.toList());
                     try {
-                        clientReference.get().deleteObjects(deleteRequest);
-                        outstanding.removeAll(keysInRequest);
-                    } catch (MultiObjectDeleteException e) {
-                        // We are sending quiet mode requests so we can't use the deleted keys entry on the exception and instead
-                        // first remove all keys that were sent in the request and then add back those that ran into an exception.
+                        DeleteObjectsResponse deleteObjectsResponse = clientReference.get().deleteObjects(deleteRequest);
                         outstanding.removeAll(keysInRequest);
-                        outstanding.addAll(
-                            e.getErrors().stream().map(MultiObjectDeleteException.DeleteError::getKey).collect(Collectors.toSet())
-                        );
-                        logger.warn(
-                            () -> new ParameterizedMessage(
-                                "Failed to delete some blobs {}",
-                                e.getErrors()
-                                    .stream()
-                                    .map(err -> "[" + err.getKey() + "][" + err.getCode() + "][" + err.getMessage() + "]")
-                                    .collect(Collectors.toList())
-                            ),
-                            e
-                        );
-                        aex = ExceptionsHelper.useOrSuppress(aex, e);
-                    } catch (AmazonClientException e) {
+                        outstanding.addAll(deleteObjectsResponse.errors().stream().map(S3Error::key).collect(Collectors.toSet()));
+                        if (!deleteObjectsResponse.errors().isEmpty()) {
+                            logger.warn(
+                                () -> new ParameterizedMessage(
+                                    "Failed to delete some blobs {}",
+                                    deleteObjectsResponse.errors()
+                                        .stream()
+                                        .map(s3Error -> "[" + s3Error.key() + "][" + s3Error.code() + "][" + s3Error.message() + "]")
+                                        .collect(Collectors.toList())
+                                )
+                            );
+                        }
+                    } catch (SdkException e) {
```

Comment:
Same, should we catch RuntimeException?

Reply:
Changed all catch points of SdkException to RuntimeException